rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -299,15 +300,12 @@ public static FlowNode getFlowEndNode(FlowExecution execution) {
     public static List<FlowNode> getChildNodes(final FlowNode parentNode) {
         final List<FlowNode> nodes = new ArrayList<FlowNode>();
 
-        FlowNodeNavigator nodeNavigator = new FlowNodeNavigator(new FlowNodeNavigationListener() {
-            @Override
-            public void onNode(FlowNode navNode) {
-                if (navNode.getParents().contains(parentNode) && !nodes.contains(navNode)) {
-                    nodes.add(navNode);
-                }
+        FlowGraphWalker walker = new FlowGraphWalker(parentNode.getExecution());
+        for (FlowNode node : walker) {
+            if (node.getParents().contains(parentNode) && !nodes.contains(node)) {
+                nodes.add(node);
```

Comment:
@tfennelly / @jglick Worth mentioning that one of the optimizations I tested was looking to see if HashSet was better for this check.  It's actually significantly worse for performance, because in almost all cases, the number of parents is 1, and creating a hashset adds an overhead vs. iterating an existing list.  The only cases where parents will be > 1 are where there's a parallel set block, and the size of the Set is still too small to merit it.